samples/memory/checkpointer-demo.ipynb

```diff
@@ -0,0 +1,486 @@
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "8eff8706-6715-4981-983b-934561ee0a19",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install general dependencies\n",
+    "from typing import Annotated, Any, Dict, List\n",
+    "\n",
+    "from langchain.chat_models import init_chat_model\n",
+    "from langchain.tools import tool\n",
+    "\n",
+    "# Import LangGraph and LangChain components\n",
+    "from langchain_core.messages import AIMessage, HumanMessage, SystemMessage\n",
+    "from langchain_core.runnables import RunnableConfig\n",
+    "from langgraph.graph import END, START, StateGraph\n",
+    "from langgraph.graph.message import add_messages\n",
+    "from langgraph.prebuilt import ToolNode, tools_condition\n",
+    "from typing_extensions import TypedDict"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "fc12fd92-b84a-4d2f-b96d-83d3da08bf70",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the AgentCoreMemorySaver that we will use as a checkpointer\n",
+    "# from langgraph_checkpoint_aws.checkpoint.agentcore_memory.saver import AgentCoreMemorySaver\n",
+    "from agentcore_memory.saver import AgentCoreMemorySaver\n",
+    "\n",
+    "import logging\n",
+    "logging.getLogger().setLevel(logging.DEBUG)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "51d91a14-541c-426c-a52c-5fdcf29f8953",
+   "metadata": {},
+   "source": [
+    "## AgentCore Memory Configuration\n",
+    "- `REGION` corresponds to the AWS region that your resources are present in, these are passed to the `AgentCoreMemorySaver`.\n",
+    "- `MEMORY_ID` corresponds to your top level AgentCore Memory resource. Within this resource we will store checkpoints for multiple actors and sessions\n",
+    "- `MODEL_ID` this is the bedrock model that will power our LangGraph agent through Bedrock Converse.\n",
+    "\n",
+    "We will use the `MEMORY_ID` and any additional boto3 client keyword args (in our case, `REGION`) to instantiate our checkpointer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "226d094c-a05d-4f88-851d-cc42ff63ef11",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "REGION = \"ap-southeast-2\"\n",
+    "MEMORY_ID = \"memory_tnpk0-AMDRU75vYn\"\n",
+    "MODEL_ID = \"apac.anthropic.claude-3-7-sonnet-20250219-v1:0\"\n",
+    "\n",
+    "# Initialize checkpointer for state persistence\n",
+    "checkpointer = AgentCoreMemorySaver(MEMORY_ID, region_name=REGION)\n",
+    "\n",
+    "# Initialize LLM\n",
+    "llm = init_chat_model(MODEL_ID, model_provider=\"bedrock_converse\", region_name=REGION)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "a8a1e31f-137a-4bd6-b959-4dc7d8176949",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define agent state and LangGraph graph builder\n",
+    "class State(TypedDict):\n",
+    "    messages: Annotated[list, add_messages]\n",
+    "\n",
```

Comment:
Rather than defining a custom graph, would prebuilt `create_react_agent` work here?
https://langchain-ai.github.io/langgraph/reference/agents/#langgraph.prebuilt.chat_agent_executor.create_react_agent

Reply:
Yes, that's a great idea. I'll implement the `create_react_agent` for this sample